src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -388,6 +392,13 @@ StartRemoteTransactionPrepare(struct MultiConnection *connection)
 
 	Assign2PCIdentifier(connection);
 
+	/* log transactions to workers in pg_dist_transaction */
```

Comment:
The current recovery logic expects every remote prepared transaction to be logged separately. We could have one record per (worker, coordinated transaction) pair, but that requires a more sophisticated implementation that matches the remote prepared transactions to a coordinated transaction ID. For metadata syncing and reference tables, such and implementation wouldn't make a difference and multi-shard commands are sufficiently rare to not worry about them. I am a bit worried about frequent COPY commands generating a ton of records, although calling recover_prepared_transactions() once a day is probably sufficient to deal with several COPY commands per second. We could also consider regressing a bit on transaction recovery here until we have an implementation that uses a more compact representation.